### PR TITLE
Support Metadata like Threadid

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

**Comment by NHDaly** [comment]
Cool, what's meta?

**Comment by vchuravy** [comment]
metadata like threadid

File: test/PProf.jl [hunk 1]
```diff
@@ -64,10 +64,10 @@ end
     end
     for i in 1:2
         if i == 1
-            data = Profile.fetch()
+            data = Profile.fetch(include_meta = true)
             args = (data,)
```
Comment (NHDaly) [inline test/PProf.jl:68]: I guess we should continue to support the old interface where you don't set `include_meta = true`?
Or would you rather require that (and assert if it's missing) and make this a breaking version change update?

**Review by NHDaly — COMMENTED** [review]
LGTM other than deciding whether we want to require include_meta if the user passes in a fetched profile. If so, we should add a test that we do throw the expected assertion.

This is really nice and simple! Sorry it took me so long to review! :) Thanks for the PR

**Comment by NHDaly** [comment]
Omg wow, literally **one year** later

File: test/PProf.jl (hunk 1 above)
Comment (NHDaly) [reply test/PProf.jl:80]: Okay I pushed up a commit, here: [7b1f044](https://github.com/JuliaPerf/PProf.jl/pull/40/commits/7b1f04406471789fb75e2cc3c79da03099edab49), for supporting non-meta profiles, but it adds a lot of complexity and i'm not sure it's right, so i'd probably just rather always require it for now. We can always relax this again in the future if we want to?

Comment (NHDaly) [reply test/PProf.jl:80]: I reverted that commit, and i think this looks good to go

**Review by NHDaly — APPROVED** [review]
Okay, I think this LGTM! I did end up adding backwards compatibility, so this should support older versions of julia as well. :)

Thanks for the PR valentin! Sorry it took me literally years to get to this. 😑 

The next thing we need to do is go harass `google/pprof` to add the ability to filter by tags in the UI!

File: test/PProf.jl (hunk 1 above)
Comment (NHDaly) [reply test/PProf.jl]: K never mind, i ended up supporting backwards compatibility. It wasn't too bad. 👍

**Comment by NHDaly** [comment]
It seems that the tests are broken on ubuntu, even on master. I'm inclined to merge this and deal with the broken tests later. Tests pass locally for me. :)

**Pull request merged**